components/led.py

```diff
@@ -234,6 +244,7 @@ def update(self) -> Hsv:
 
     def pick_new_message(self) -> None:
         # QUESTION? Should functions take args or assume previous step already done
+        self.start_time = self.clock()  # Reset the time of the pattern
```

Comment:
Does this still also need to happen in `__post_init__` if this method is being called there anyway?